### DISCUSSION TITLE
Deck gl with Jest mock WebMercatorViewport object

### DISCUSSION THREAD

**Discussion by ariel-upstream** [visgl/deck.gl#8809] [Q&A]
Hi, I'm trying to mock the object of WebMercatorViewport with the fitBounds function.
but I can't get rid of the error: 
```
(intermediate value).fitBounds is not a function
TypeError: (intermediate value).fitBounds is not a function
```

In my setupTests I'm using jest mock:

```
  jest.mock("@deck.gl/core", () => ({
  __esModule: true,
  WebMercatorViewport: jest.fn(() => ({
    fitBounds: jest.fn(),
  })),
  default: PropsViewer("DeckGL"),
}))
```

**Comment by ibgreen** [comment]
Sounds like you just need to bind your mock fitbounds method. Consult you jest documentation.

**Reply by ariel-upstream** [reply]
Hi thanks, Didn't find something like that. do you have any links related to it?
